lib/argument_deduplication/client.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+module ArgumentDeduplication
+  class Client
+    include Sidekiq::ClientMiddleware
+
+    def call(_worker, job, _queue, _redis_pool)
+      process_arguments!(job)
+      yield
+    end
+
+    private
+
+    def process_arguments!(job)
+      return unless job['deduplicate_arguments']
```

Comment:
One thing I failed to catch is that in this current form I think it makes migrating a live server impossible, as it will misunderstand the arguments for enqueued jobs.

Reply:
The simplest way to do that is probably to just create a new class for those jobs that inherits from them.